### ISSUE TITLE
[Feature Request] Family sharing indication (game owned by a family member) on store pages

### ISSUE THREAD

**Issue by enchained** [tfedor/AugmentedSteam#322]
I've set up Steam family sharing, so whenever I want to get the game on sale, I'm wondering whether my boyfriend already has it. But there's no indication on the store pages whether the game is available for me. And it could be useful to have install/play button just like with the owned games on such pages. Of course, there is "friends who already own this game" list, but since I have >6 friends, on popular games I often have to open a separate page to see him listed.

I'm not sure if any family sharing info is accessible from the web, probably it's just a client library data. But this got me thinking, it would be nice to have specific friend(s) pinned to the top of the "friends who already own this game" list, so they'll be visible right away. Also can be useful if you have just a few closest friends and want to be able to always see them in the store when looking for gifts.

**Comment by Revadike** [comment]
This could work, if there is some way to find who is sharing their library with you, but I can't find this information anywhere.
There is `https://api.steampowered.com/IPlayerService/IsPlayingSharedGame/v1?key=APIKEY&steamid=STEAMID&appid_playing=APPID`, but that only works when you're ingame.
Then, there's https://partner.steamgames.com/doc/webapi/ISteamUser#CheckAppOwnership, which sounds great, but you need a publisher api key for this.

**Comment by enchained** [comment]
Yeah, looks like no way to automate it yet. I hoped that with the new steam library view it'll come to the web and we'll have more data, but it looks like it's not something in their plans. I guess the only way to implement it for now is to point users from friends list manually, import their games list, and exclude games with family sharing disabled. Not sure if a viable idea. 

And there's still "friends pinning" idea that can be a temporary "fix". Maybe even use the favorite friends data from the steam chat, if there's any access to that.

**Comment by Revadike** [comment]
> I hoped that with the new steam library view it'll come to the web

I actually looked into that. It only uses these endpoints:
https://store.steampowered.com/broadcast/ajaxgetappinfoforcap?appid=729570&feature=library-partner-events&cc=NL&l=english
and https://store.steampowered.com/api/libraryappdetails/?appid=729570&l=english

**Comment by Revadike** [comment]
Actually yeah, a friend stalker/pinning mode could work.

**Comment by Revadike** [comment]
So, you can select certain friends to pin and then on every steam store page, it shows if these friends own the game or not.
The friendsthatplay page can be used to get this info .
It could look like this:
![](https://media.discordapp.net/attachments/540532923657617415/679038876085387523/unknown.png)

@MxtOUT What do you think?

**Comment by MxtOUT** [comment]
What if you don't own the game?
Either way, this is too unobtrusive, it needs to stand out more

**Comment by Revadike** [comment]
> What if you don't own the game?
> Either way, this is too unobtrusive, it needs to stand out more

how I imagined it: it would show the same "in library" block, but without "your"
Anyway, any other ideas?

**Comment by MxtOUT** [comment]
Somewhere here?
![image](https://user-images.githubusercontent.com/14999077/74860864-ab4eaf00-5349-11ea-9ce2-ff972e267df8.png)

**Comment by candela97** [comment]
Since the new Steam Families released, there's now a banner like this on eligible apps:
![螢幕擷取畫面 2024-09-16 025408](https://github.com/user-attachments/assets/e4372723-48ed-42dd-ae1f-39e3e3e4c5c7)
So I guess this is resolved?

Pinned friends sounds neat, but iirc there're were complaints from Steam when the old ES called the `appuserdetails` endpoint too frequently 😅

**Comment by agharbeia** [comment]
> Since the new Steam Families released, there's now a banner like this on eligible apps:

and also regarding @enchained's comment: 

> https://store.steampowered.com/account/managedevices only shows who YOU are sharing with and other irrelevant information.

The new Family sharing system means that sharing is mutual. You share with, and only with, those who share with you. So the requested functionality could be achievable, now.

#2068 is a Duplicate of this, by the way.

**Comment by Rasmal2023** [comment]
@candela97 @agharbeia Yes each game will show this indicator when you enter its page
it would be much better if we can see this highlighted in the list as well if possible

**Comment by enchained** [comment]
> highlighted in the list as well if possible

SteamDB extension already [does this](https://github.com/SteamDatabase/BrowserExtension/blob/189ec45ebb82ac5b971c65eb7efdc459d1f9de90/scripts/background.js#L227), but the [GetSharedLibraryApps](https://steamapi.xpaw.me/#IFamilyGroupsService/GetSharedLibraryApps) endpoint requires getting some other token instead of API key, despite what generated docs say. It should work for other extensions like AS, _but not so useful for independent steam tools since you have to be logged-in in order to get the token._

And it might be nice if AS could add something like the "in ITAD collection" bar, but mentioning a Steam Family collection, for a much better visibility. Or even sync those to the ITAD collection with a specific family tag or something.

Also, for some people (i.e. who live and play together on the same PC) it might not matter whether the game has family sharing enabled or not. When it is disabled, they can still play from account who owns it, but there's still no official way to highlight it on the store page for those families. I think there are two ways to solve it: 
1. Keeping a family usernames list + looking for username in the friends that own page (might have issues with API rate limiting)
2. Keeping a family usernames list + fetching their libraries via API on a schedule (public profile required, no DLC info)

Friends pinning still might be relevant for prioritizing friends in cases like "seeing if X also owns this co-op game", but not sure if it's worth creating a separate ticket due to API rate issues. Maybe fetching specific friends libraries from (2) above could help with friends pinning too?

**Comment by Rasmal2023** [comment]
True that steamdb does that 
but it's done on the steamdb website 

![Image](https://github.com/user-attachments/assets/15a038b6-2632-4b99-8c5e-b3e7f31b8bfe)

as you can see it shows wish listed games and family shared games 
is it possible to take this into the store itself since it's already showing wish listed games and owned games

**Comment by enchained** [comment]
> True that steamdb does that
> but it's done on the steamdb website

I did not mean that steamdb already does that for the steam store, I just mentioned the possibility, since there is a way to get that data for AS highlighting feature.

**Comment by GeorgeGedox** [comment]
Could such an indicator be added on the wishlist page too?

**Comment by c26zwj** [comment]
I would love this feature too
